src/test/java/seedu/contax/logic/commands/FindAppointmentCommandTest.java

```diff
@@ -0,0 +1,112 @@
+                new NameContainsKeywordsPredicate(Collections.singletonList("first"));
+        NameContainsKeywordsPredicate secondPredicate =
+                new NameContainsKeywordsPredicate(Collections.singletonList("second"));
+
+        FindAppointmentCommand findFirstCommand = new FindAppointmentCommand(firstPredicate);
+        FindAppointmentCommand findSecondCommand = new FindAppointmentCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FindAppointmentCommand findFirstCommandCopy = new FindAppointmentCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void clientName_equals() {
+        PersonNameContainsKeywordsPredicate firstPredicate =
+                new PersonNameContainsKeywordsPredicate(Collections.singletonList("client1"));
+        PersonNameContainsKeywordsPredicate secondPredicate =
+                new PersonNameContainsKeywordsPredicate(Collections.singletonList("client2"));
+
+        FindAppointmentCommand findFirstCommand = new FindAppointmentCommand(firstPredicate);
+        FindAppointmentCommand findSecondCommand = new FindAppointmentCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FindAppointmentCommand findFirstCommandCopy = new FindAppointmentCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different keyword -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_nonMatchingNameKeywords_noAppointmentFound() {
+        Model model = new ModelManager(getTypicalAddressBook(), new Schedule(), new UserPrefs());
+        Model expectedModel = new ModelManager(getTypicalAddressBook(), new Schedule(), new UserPrefs());
+        CommandResult expectedResult = new CommandResult(String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 0),
+                GuiListContentType.APPOINTMENT);
+        model.addAppointment(new AppointmentBuilder(APPOINTMENT_ALICE).build());
+        expectedModel.addAppointment(new AppointmentBuilder(APPOINTMENT_ALICE).build());
+        NameContainsKeywordsPredicate predicate = new NameContainsKeywordsPredicate(Arrays.asList("bob", "charlie"));
```

Comment:
You are missing the `PersonNameContainsKeywordPredicate` test, and you seem to have confused the 2 predicates.

Reply:
Added